### PR TITLE
Update functions

### PR DESCRIPTION
**Pull request by ewoutp**
This PR adds `Update...` functions for all resources.

To make this work, I had to change the protoc generator to `gogofast_out`. 
As a result lots of types have changed to pointer types.
This makes the library harder to work with. To compensate (a bit), I've added helper functions to convert primitive types to their pointer versions (e.g. `StringP("hello world")`) and back (e.g. `String(strPtr)`.

In the process, I've updated the kubernetes version to 1.5.1.

### CODE REVIEW

File: gen.sh [hunk 1]
```diff
@@ -21,16 +21,18 @@ cd assets
...
+    protoc --gogofast_out=$PKG $file
 done
 
 cd -
 
 mv k8s.io/kubernetes/pkg/* .
 rm -rf k8s.io
-sed -i 's|"k8s.io/kubernetes/pkg|"github.com/ericchiang/k8s|g' $( find {api,apis,util,runtime} -name '*.go' )
-sed -i 's|"k8s.io.kubernetes.pkg.|"github.com/ericchiang.k8s.|g' $( find {api,apis,util,runtime} -name '*.go' )
+sed -i '' 's|"k8s.io/kubernetes/pkg|"github.com/ericchiang/k8s|g' $(find {api,apis,util,runtime} -name '*.go')
```
Comment (ewoutp) [inline gen.sh]: This is to make it work on OSX.

File: gen.sh [hunk 2]
```diff
@@ -21,16 +21,18 @@ cd assets
...
+sed -i '' 's|"k8s.io.kubernetes.pkg.|"github.com/ericchiang.k8s.|g' $(find {api,apis,util,runtime} -name '*.go')
 
 rm -rf assets
 
-go run gen.go
+cd .gobuild/src/github.com/ericchiang/k8s && go run gen.go
+cd -
```
Comment (ewoutp) [inline gen.sh]: This is for my build environment and should not be included in final merge.

**Comment by ericchiang** [comment]
Being able to detect the sub-resources is the harder part here. I don't see this address in this proposed patch.

**Comment by ewoutp** [comment]
What exactly do you mean by that?

**Comment by ericchiang** [comment]
ah woops, I was confused with patch.

**Comment by ewoutp** [comment]
Ok got it.
No this is indeed about "replace".
I considered calling the methods `Replace...` but for similarity with the methods for kubernetes/client-go I kept it `Update...`

**Comment by ericchiang** [comment]
Thanks, though I ended up implementing this myself.

**Pull request closed without merging**